VisualPinball.Engine/VPT/Bumper/Bumper.cs

```diff
@@ -8,6 +8,7 @@ public class Bumper : Item<BumperData>, IRenderable, IHittable
 	{
 		public bool IsCollidable => true;
 		public EventProxy EventProxy { get; private set; }
+		public string[] UsedMaterials => new string[] { Data.CapMaterial, Data.BaseMaterial, Data.RingMaterial, Data.SocketMaterial };

```

Comment:
I assume this is called by the editor loop? No issue with garbage collection if we instantiate it each time? Maybe put it on a readonly prop and just return it?

Reply:
No its not called regularly, just when the window's rows need to populate to get the "in use" flag and when the mat gets edited to dirty the items. If we make a persistent field for it we'd have to also add a dirty flag or some update call to rebuild the list when the item changes. I was trying to keep things simple, but let me think on it, see if we can build a simple pattern for all that.

Reply:
If it's not called every frame, that's fine then.